### PR TITLE
websocket: avoid lock while calling handler callback

### PR DESCRIPTION
**Pull request by safchain**

### CODE REVIEW

**Comment by safchain** [comment]
run skydive-cdd-overview-tests
run skydive-compile-tests
run skydive-functional-hw-tests
run skydive-functional-tests-backend-elasticsearch

**Comment by safchain** [comment]
run skydive-cdd-overview-tests
run skydive-compile-tests
run skydive-functional-hw-tests
run skydive-functional-tests-backend-elasticsearch

**Comment by safchain** [comment]
run skydive-cdd-overview-tests
run skydive-functional-hw-tests
run skydive-functional-tests-backend-elasticsearch

**Comment by safchain** [comment]
run skydive-functional-hw-tests
run skydive-functional-tests-backend-elasticsearch

**Comment by safchain** [comment]
run skydive-functional-tests-backend-elasticsearch

File: websocket/client.go [hunk 1]
```diff
@@ -320,6 +320,14 @@ func (c *Conn) Start() {
 	go c.run()
 }
 
+func (c *Conn) cloneEventHandlers() (handlers []SpeakerEventHandler) {
+	c.RLock()
+	handlers = append(handlers, c.eventHandlers...)
```
Comment (nplanel) [inline websocket/client.go]: should be Lock() as you writing, only one instance should have access to this ressource

File: websocket/message.go [hunk 2]
```diff
@@ -264,13 +264,14 @@ func (a *structSpeakerEventDispatcher) dispatchMessage(c *StructSpeaker, m *Stru
 	}
 
 	a.eventHandlersLock.RLock()
```
Comment (nplanel) [inline websocket/message.go]: same here

File: websocket/pool.go [hunk 3]
```diff
@@ -68,26 +68,30 @@ type incomerPool struct {
 	*Pool
 }
 
+func (s *Pool) cloneEventHandlers() (handlers []SpeakerEventHandler) {
+	s.eventHandlersLock.RLock()
```
Comment (nplanel) [inline websocket/pool.go]: here

**Comment by safchain** [comment]
run skydive-functional-tests-backend-elasticsearch

File: websocket/client.go (hunk 1 above)
Comment (lebauce) [reply websocket/client.go]: there is no writing to c.eventHandlers here, only read/copy

**Pull request merged**